src/components/carousel/index.jsx

```diff
@@ -170,6 +170,26 @@ const Carousel = ({
 		Children.map(children, (child) => (child?.type === subcomponentType ? child : null))
 	);
 
+	const childItems = Children.toArray(subComponents);
+
+	let carouselItems = childItems.map((child, index) => {
+		const viewable = index + 1 > slide - slidesToShowInView && index + 1 <= slide;
+		return child.type === Item ? cloneElement(child, { viewable }) : null;
+	});
+
+	const totalSlides = carouselItems.length;
+
+	const emitEvent = (eventName, page, options) => {
```

Comment:
This should be wrapped in a `useCallback` hook, with an appropriate dependency array. I noticed some weirdness when testing the enter/exit fullscreen events (it was getting called more times each time the gallery opened/closed) and this seems to be the cause.